### PR TITLE
Features/#332 allocation pumped hydro

### PR DESCRIPTION
**Pull request by IlkaCu**
Implements the allocation of pumped hydro capacities.

### CODE REVIEW

**Comment by IlkaCu** [comment]
Code was successfully tested on CI branch.

File: src/egon/data/airflow/dags/pipeline.py [hunk 1]
```diff
@@ -457,6 +458,7 @@
             data_bundle,
             zensus_mv_grid_districts,
             district_heating_areas,
+            power_plants,
```
Comment (ClaraBuettner) [inline src/egon/data/airflow/dags/pipeline.py:461]: The HeatSupply-Dataset doesn't have a dependency to the PowerPlants anymore. So this line could be removed.

File: src/egon/data/datasets/chp/__init__.py [hunk 2]
```diff
@@ -52,6 +52,21 @@ class EgonMaStRConventinalWithoutChp(Base):
...
+            dependencies=dependencies,
+            tasks=(
+                create_tables,
+                insert_chp_egon2035,
+                assign_heat_bus,
+                # extension
+            ),
+        )
+
+
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/chp/__init__.py:69]: These lines are also not used anymore, looks a bit like a merge problem.

File: src/egon/data/datasets/storages/pumped_hydro.py [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    unmatched: pandas.DataFrame
+        Units for which no geolocation could be identified
+
+    located : pandas.DataFrame
+        Units with a geolocation based on their city information
+
+    """
+
+    # import geolocator
+    from geopy.geocoders import Nominatim
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/storages/pumped_hydro.py:278]: Why is this package only imported in this function and not in the beginning of the script?

File: src/egon/data/airflow/dags/pipeline.py (hunk 1 above)
Comment (IlkaCu) [reply src/egon/data/airflow/dags/pipeline.py]: resolved by removing this dependency.

File: src/egon/data/datasets/storages/pumped_hydro.py (hunk 3 above)
Comment (IlkaCu) [reply src/egon/data/datasets/storages/pumped_hydro.py]: Good question and no answer. ;-) I moved it

**Pull request merged**